docs/ci/codemagic.md

```diff
@@ -0,0 +1,99 @@
+---
+sidebar_position: 2
+title: 🚀 Codemagic
+description: Integrate Shorebird into your Codemagic workflow
+---
+
+# Codemagic Workflow Integration
+
+This guide will help you integrate Shorebird into your Codemagic Workflow using the Codemagic Workflow Editor and YAML configuration.
+
+## Prerequisites
+
+✅ Shorebird CLI is installed on your machine
+
+✅ You are logged into a paid account.
+
+:::info
+Refer to the [getting started](/) instructions for more information.
+:::
+
+## Quick Start
+
+You can integrate Shorebird into your CI either using the Codemagic Workflow editor or using YAML.
+
+## Authentication
+
+Most Shorebird functionality, like creating releases and patches, requires being authenticated. In order to authenticate with Shorebird in CI, you will need to generate a CI token.
+
+```sh
+shorebird login:ci
+```
+
+You will be prompted to go through a similar OAuth Flow as when using `shorebird login`, however, `shorebird login:ci` will not store any credentials on your device. Instead, a Shorebird token will be generated for you to use in CI.
+
+The output should look something like:
+
+```sh
+$ shorebird login:ci
+The Shorebird CLI needs your authorization to manage apps, releases, and patches on your behalf.
+
+In a browser, visit this URL to log in:
+
+https://accounts.google.com/o/oauth2/v2/auth...
+
+Waiting for your authorization...
+
+🎉 Success! Use the following token to login on a CI server:
+
+<SHOREBIRD_TOKEN>
+
+Example:
+
+export SHOREBIRD_TOKEN="$SHOREBIRD_TOKEN" && shorebird patch android
+```
+
+:::caution
+The `SHOREBIRD_TOKEN` is a secret and should not be committed directly in your source code or shared publicly.
+:::
+
+Next, copy the generated `SHOREBIRD_TOKEN` and navigate to your Codemagic secrets via:
+
+1. Go to "Environment Variables"
+2. Enter as variable name `SHOREBIRD_TOKEN`
+3. Paste the token into the variable value field
+4. Select or create new group, like "shorebird"
+
+### Codemagic Workflow Editor
+
+1. Go to the "Environment variables" section of your application settings.
+
+2. Add a new environment variable named `SHOREBIRD_TOKEN` and paste your token as the value.
+
+3. Save the changes.
+
+### Codemagic YAML
+
+In the `codemagic.yaml` file, specify the `SHOREBIRD_TOKEN` under `environment`.
+
+```yaml
+workflows:
+  shorebird-workflow:
+    name: Shorebird Workflow
+    environment:
+      vars:
+        SHOREBIRD_TOKEN: Encrypted($TOKEN)
+    scripts:
+      - name: Install Shorebird
+        script: brew install shorebird
```

Comment:
I think no brew package found for shorebird

<img width="1023" alt="Screenshot 2023-07-05 at 2 04 10 PM" src="https://github.com/shorebirdtech/docs/assets/131065273/0b81ad84-860b-446c-ae93-1a6489e9776b">

Reply:
Yes, it's the command does not exist. 75% of this PR is generated with ChatGPT and hadn't the time to complete the docs. This line wrote ChatGPT and isn't true 😅

Reply:
but it's truly needed feature, thank you for taking the time doing it anyway, and hope to be completed and merged 🙏🏼